mixer/pkg/perf/run.go

```diff
@@ -130,3 +141,59 @@ func run(b benchmark, setup *Setup, env *Env, executablePathSuffix string, copro
 	// graceful close of clients during teardown.
 	controller.close()
 }
+
+func runDispatcherOnly(b benchmark, setup *Setup, settings *Settings) {
+	a, err := initializeArgs(settings, setup)
+	if err != nil {
+		b.fatalf("error initializing args: '%s'", err.Error())
+		return
+	}
+
+	s, err := mixer.New(a)
+	if err != nil {
+		b.fatalf("error creating Mixer server: '%s'", err.Error())
+		return
+	}
+	defer s.Close()
+
+	dispatcher := s.GetDispatcher()
+
+	list := attribute.GlobalList()
+	globalDict := make(map[string]int32, len(list))
+	for i := 0; i < len(list); i++ {
+		globalDict[list[i]] = int32(i)
+	}
+
+	requests := setup.Load.createRequestProtos(setup.Config)
+	bags := make([]attribute.Bag, len(requests)) // precreate bags to avoid polluting allocation data.
+	for i, r := range requests {
+		switch r.(type) {
+		case *istio_mixer_v1.ReportRequest:
+			rrep := r.(*istio_mixer_v1.ReportRequest)
+			bags[i] = attribute.NewProtoBag(&rrep.Attributes[0], globalDict, attribute.GlobalList())
+
+		case *istio_mixer_v1.CheckRequest:
+			crep := r.(*istio_mixer_v1.CheckRequest)
+			bags[i] = attribute.NewProtoBag(&crep.Attributes, globalDict, attribute.GlobalList())
+
+		default:
+			b.fatalf("unknown request type: %v", r)
+		}
+	}
+
+	b.run("DispatcherOnly", func(bb benchmark) {
+		for i := 0; i < bb.n(); i++ {
+			for j, r := range requests {
+				bag := bags[j]
+
+				switch r.(type) {
+				case *istio_mixer_v1.ReportRequest:
+					_ = dispatcher.Report(context.Background(), bag)
```

Comment:
Don't you want to track errors? You don't want unreported errors skewing benchmark numbers...